.github/workflows/release-on-pr-merge-to-main.yml

```diff
@@ -0,0 +1,133 @@
+name: Build a release and increment SemVer upon pr/merge to main branch
```

Comment:
I'm not gonna test this out right now - I'll assume it works. We can test it the next time we need to make an update!